src/BinOp.chpl

```diff
@@ -128,6 +129,55 @@ module BinOp
     return res;
   }
 
+  //
+  // TODO: these checks sets are used only to check if a string matches
+  // i.e. realOps.contains(op) or boolOps.contains(op)
+  // however, a faster way to do this is to write a function that checks
+  // the ascii bytes
+  // for example, the boolOps can be written as:
+  //
+  // return (op.numBytes == 1 && (op.byte[0] == 60 ||
+  //                             op.byte[0] == 62)) ||
+  //        (op.numBytes == 2 && ((op.byte[0] == 60 && op.byte[1] == 61) ||
+  //                              (op.byte[0] == 62 && op.byte[1] == 61) ||
+  //                              (op.byte[0] == 61 && op.byte[1] == 61) ||
+  //                              (op.byte[0] == 33 && op.byte[1] == 61)));
+  //
+  // it should also be possible to use the `toByte` method at compile-time to
+  // improve the readability, although this has not been tested to see how it impacts
+  // compile-time and runtime performance:
```

Comment:
If the worry about the compile time hit by this is because we will see this kind of `return`s all over the place, we can always have `param lessThan = "<".toByte();` etc. at module-scope to compile `toByte` functions only once.

In general, do you know empirically that either of these approaches makes things faster. I'd be surprised if `set.contains` is taking up too much time, even with small data sizes.

Reply:
Oh, I see that in the discussion above with Ryan you mention that they do indeed make the code go faster.